src/QueryBuilder.php

```diff
@@ -113,6 +97,67 @@ public function getSearchParams(string $queryString, string $type, int $start, i
         return $params;
     }
 
+    /**
+     * @param string[] $blocks
+     */
+    private function buildConditions(string $index, array $blocks): array
+    {
+        $conditions = $this->buildConditionsByType($index, $blocks);
+
+        if ($this->includeAccessData($index)) {
+            $conditions[] = $this->buildAccessConditions();
+        }
+
+        return $conditions;
+    }
+
+    /**
+     * Use only simple input
+     * @param string[] $blocks
+     * @return string[]
+     */
+    private function getResultsConditions(array $blocks): array
+    {
+        $conditions = [];
+
+        foreach ($blocks as $block) {
+            $block = $this->parseBlock($block);
+            if ($block->getField() === 'parent_classes') {
+                continue;
+            }
+
+            $conditions[] = sprintf('("%s")', $block->getTerm());
+        }
+
+        return $conditions;
+    }
+
+    private function buildConditionsByType(string $type, array $blocks): array
+    {
+        if ($type === IndexerInterface::DELIVERY_RESULTS_INDEX) {
+            return $this->getResultsConditions($blocks);
+        }
+
+        return $this->getResourceConditions($blocks);
+    }
+
+    private function getResourceConditions($blocks): array
+    {
+        foreach ($blocks as $block) {
+            $queryBlock = $this->parseBlock($block);
+
+            if (empty($queryBlock->getField())) {
+                $conditions[] = sprintf('("%s")', $queryBlock->getTerm());
+            } elseif ($this->isStandardField($queryBlock->getField())) {
```

Comment:
could we bind these in order to reduce the complexity

Reply:
or refactor this ?

Reply:
I would not like to refactor this under this PR. We can create new PR for this. It is already to many changes here.